app/models/user.rb

```diff
@@ -9,4 +9,39 @@ class User < ApplicationRecord
   has_many :posts
   has_many :comments, dependent: :destroy
   has_many :likes, dependent: :destroy
+  has_many :friendships # sent requests
+  has_many :inverse_friendships, class_name: 'Friendship', foreign_key: 'friend_id' # received requests
+
+  has_many :pending_friendships, -> { where(confirmed: false) }, class_name: 'Friendship', foreign_key: 'user_id'
+  has_many :pending_friends, through: :pending_friendships, source: :friend_id
+
+  has_many :accepted_friendships, -> { where(confirmed: true) }, class_name: 'Friendship'
+  has_many :friends, through: :accepted_friendships, source: :friend_id
+
+  has_many :inverse_friendships, -> { where(confirmed: false) }, class_name: 'Friendship', foreign_key: 'friend_id'
+  has_many :friend_requests, through: :inverse_friendships, source: :user_id
```

Comment:
- [ ] Remove these two lines of SQL queries. They'll come in handy in the next milestone. The ruby enumerables will do the job perfectly for now.